packages/runtime/src/interpreter/opcode-list.ts

```
@@ -4355,21 +4360,44 @@ export class ITxn extends Op {
 			});
 		}
 
-		let result;
-		// what is "last "
+		let result: StackElem;
+		// what is "last"
 		const groupTx = this.interpreter.innerTxnGroups[this.interpreter.innerTxnGroups.length - 1];
 		const tx = groupTx[groupTx.length - 1];
 
 		switch (this.field) {
 			case "Logs": {
-				// TODO handle this after log opcode is implemented
-				// https://www.pivotaltracker.com/story/show/179855820
-				result = 0n;
+				if (
+					this.interpreter.tealVersion >= 5 &&
+					this.idx !== undefined &&
+					this.interpreter.mode == ExecutionMode.APPLICATION
+				) {
+					// handle Logs
+					const txReceipt = this.interpreter.runtime.ctx.state.txReceipts.get(tx.txID);
+					const logs: Uint8Array[] = txReceipt?.logs ?? [];
+					this.checkIndexBound(this.idx, logs, this.line);
+					result = logs[this.idx];
+				} else {
+					result = 0n;
```

Should we throw an error when there is no result?